coordinator/internal/orm/batch.go

```diff
@@ -190,21 +190,73 @@ func (o *Batch) InsertBatch(ctx context.Context, batch *encoding.Batch, dbTX ...
 		return nil, errors.New("invalid args: batch is nil")
 	}
 
+	numChunks := uint64(len(batch.Chunks))
+	if numChunks == 0 {
+		return nil, errors.New("invalid args: batch contains 0 chunk")
+	}
+
 	daBatch, err := codecv0.NewDABatch(batch)
 	if err != nil {
 		log.Error("failed to create new DA batch",
 			"index", batch.Index, "total l1 message popped before", batch.TotalL1MessagePoppedBefore,
-			"parent hash", batch.ParentBatchHash, "number of chunks", len(batch.Chunks), "err", err)
+			"parent hash", batch.ParentBatchHash, "number of chunks", numChunks, "err", err)
 		return nil, err
 	}
 
+	var startChunkIndex uint64
+	parentBatch, err := o.GetLatestBatch(ctx)
+	if err != nil && !errors.Is(errors.Unwrap(err), gorm.ErrRecordNotFound) {
```

Comment:
Do we need to unwrap here?

Reply:
GetLatestBatch would not return `gorm.ErrRecordNotFound` error:
https://github.com/scroll-tech/scroll/blob/a42f8106b8eedbb512691def98119cad5f7f9fa3/rollup/internal/orm/batch.go#L139-L141

thus removed `errors.Is(errors.Unwrap(err), gorm.ErrRecordNotFound)` check.

also changed `getLatestChunk` and `InsertChunk`'s error handling.

related commit: https://github.com/scroll-tech/scroll/pull/1196/commits/d8ae98707fad2af4b5e5986fc214ae48776b2734